pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -519,57 +519,52 @@ protected void internalGetPendingAckInternalStats(AsyncResponse asyncResponse, b
 
                         if (e != null) {
-                            asyncResponse.resume(new RestException(e));
+                            resumeAsyncResponseExceptionally(asyncResponse, e);
                             return;
                         }
                         if (!optionalTopic.isPresent()) {
-                            asyncResponse.resume(new RestException(TEMPORARY_REDIRECT,
-                                    "Topic is not owned by this broker!"));
+                            asyncResponse.resume(new RestException(NOT_FOUND, "Topic not found"));
                             return;
                         }
                         Topic topicObject = optionalTopic.get();
-                        if (topicObject instanceof PersistentTopic) {
-                            try {
-                                ManagedLedger managedLedger =
-                                        ((PersistentTopic) topicObject).getPendingAckManagedLedger(subName).get();
-                                TransactionPendingAckInternalStats stats =
-                                        new TransactionPendingAckInternalStats();
-                                TransactionLogStats pendingAckLogStats = new TransactionLogStats();
-                                pendingAckLogStats.managedLedgerName = managedLedger.getName();
-                                pendingAckLogStats.managedLedgerInternalStats =
-                                        managedLedger.getManagedLedgerInternalStats(metadata).get();
-                                stats.pendingAckLogStats = pendingAckLogStats;
-                                asyncResponse.resume(stats);
-                            } catch (Exception exception) {
-                                if (exception instanceof ExecutionException) {
-                                    if (exception.getCause() instanceof ServiceUnitNotReadyException) {
-                                        asyncResponse.resume(new RestException(SERVICE_UNAVAILABLE,
-                                                exception.getCause()));
-                                        return;
-                                    } else if (exception.getCause() instanceof NotAllowedException) {
-                                        asyncResponse.resume(new RestException(METHOD_NOT_ALLOWED,
-                                                exception.getCause()));
-                                        return;
-                                    } else if (exception.getCause() instanceof SubscriptionNotFoundException) {
-                                        asyncResponse.resume(new RestException(NOT_FOUND, exception.getCause()));
-                                        return;
-                                    }
+                        try {
+                            ManagedLedger managedLedger =
+                                    ((PersistentTopic) topicObject).getPendingAckManagedLedger(subName).get();
```

Comment:
is this a async call ?  in this case we should use "thenCompose" and concatenate the async calls

@mattisonchao FYI

Reply:
Ok. I decided to change it to async, but a little from the original idea.
Anyway, we should avoid using it like this. 
I will update later.

Reply:
@eolivelli